src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -137,12 +137,32 @@ public class ReleaseService : IReleaseService
                 });
         }
 
-        public Task<Either<ActionResult, bool>> DeleteReleaseAsync(Guid releaseId)
+        public Task<Either<ActionResult, Unit>> DeleteRelease(Guid releaseId)
         {
             return _persistenceHelper
                 .CheckEntityExists<Release>(releaseId)
                 .OnSuccess(_userService.CheckCanDeleteRelease)
-                .OnSuccess(async release =>
+                .OnSuccessDo(async () =>
+                {
+                    // Delete any replacements that might exist
+                    var filesWithReplacements = await _context.ReleaseFiles
```

Comment:
Actually do we delete any of the amendment files? It might not just be replacement files?

Reply:
Good point, just checked and that's true, and it's causing another bug related to replacing data on amendments that I've just found while testing this so I will make this change! 🥇

Reply:
Cool, so this change turns out to be next to nothing except for removing those ReleaseFile links to soft deleted Releases and removing ReleaseFileReferences if those files are then orphaned (i.e. because they were uploaded to the amendment) 😃

Reply:
A new issue has been opened to clean up these files that have been retained historically and physically delete whole Releases rather than soft deleting them in future.